core/src/SYCL/Kokkos_SYCL_Parallel_Scan.hpp

```diff
@@ -32,21 +32,22 @@ namespace Impl {
 template <int dim, typename ValueType, typename FunctorType>
 void workgroup_scan(sycl::nd_item<dim> item, const FunctorType& final_reducer,
                     sycl::local_accessor<ValueType> local_mem,
-                    ValueType& local_value, unsigned int global_range) {
+                    ValueType& local_value, int global_range) {
   // subgroup scans
-  auto sg                = item.get_sub_group();
-  const auto sg_group_id = sg.get_group_id()[0];
-  const auto id_in_sg    = sg.get_local_id()[0];
-  for (unsigned int stride = 1; stride < global_range; stride <<= 1) {
+  auto sg               = item.get_sub_group();
+  const int sg_group_id = sg.get_group_id()[0];
+  const int id_in_sg    = sg.get_local_id()[0];
+
+  for (int stride = 1; stride < global_range; stride <<= 1) {
     auto tmp = sg.shuffle_up(local_value, stride);
     if (id_in_sg >= stride) final_reducer.join(&local_value, &tmp);
   }
 
-  const auto max_subgroup_size = sg.get_max_local_range()[0];
-  const auto n_active_subgroups =
+  const int max_subgroup_size = sg.get_max_local_range()[0];
+  const int n_active_subgroups =
       (global_range + max_subgroup_size - 1) / max_subgroup_size;
 
-  const auto local_range = sg.get_local_range()[0];
+  const int local_range = sg.get_local_range()[0];
```

Comment:
did SYCL change so that you need int now?

Reply:
No, it's not necessary to do that but I have seen that SYCL is very sensitive to 64-bit index calculations, and local indices surely will never require that.

Reply:
Just to be clear, `auto` deduces `size_t`, not `int`.  I agree we don't need that much of an index range.